internal/cluster/kube/deployment.go

```diff
@@ -0,0 +1,145 @@
+	}
+
+	return deploys.Items, nil
+}
+
+func scaleUpDeployment(clientSet kubernetes.Interface, name, namespace string, timeout int) error {
+	deploy, err := getDeployment(clientSet, name, namespace)
+	if err != nil {
+		logger.Errorf(err, "Could not get the deployment %s.%s", name, namespace)
+		return err
+	}
+
+	deploy.Spec.Replicas = pointer.Int32Ptr(1) // TODO make this configurable
+
+	if _, err := updateDeployment(clientSet, deploy, namespace); err != nil {
+		logger.Errorf(err, "Could not scale up the deployment %s.%s", name, namespace)
+		return err
+	} else {
+		return waitForDeploymentAvailable(clientSet, name, namespace, timeout)
+	}
+}
+
+func waitForDeploymentAvailable(clientSet kubernetes.Interface, name, namespace string, timeout int) error {
+	err := wait.PollImmediate(time.Second, time.Duration(timeout)*time.Second, func() (bool, error) {
+		if deploy, err := getDeployment(clientSet, name, namespace); err != nil {
+			logger.Errorf(err, "Could not get the deployment %s.%s", name, namespace)
+			return true, err
+		} else {
+			if deploy.Status.AvailableReplicas >= 1 { // TODO make this configurable
+				logger.Debugf("Deployment %s.%s scaled up successfully", name, namespace)
+				return true, nil
+			} else {
+				logger.Debugf("Deployment %s.%s not ready yet", name, namespace)
+			}
+			return false, nil
+		}
+	})
+	return err
+}
+
+func scaleDownDeployments(
+	kubeClient kubernetes.Interface,
+	namespace string,
+	mustScaleDown func(deployName, namespace string) (bool, error),
+) []error {
+	labelSelector := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", cluster.LabelDeploymentProxless, "true"),
+	}
+
+	deploys, err := listDeployments(kubeClient, namespace, labelSelector)
+
+	var errs []error
+	if err != nil {
+		logger.Errorf(
+			err,
+			"Could not list deployments with label %s in namespace %s",
+			labelSelector.LabelSelector, namespace)
+		errs = append(errs, err)
+	} else {
+		for _, deploy := range deploys {
+			if *deploy.Spec.Replicas > int32(0) {
+				scaleDown, _ := mustScaleDown(deploy.Name, deploy.Namespace)
+				if scaleDown {
+					deploy.Spec.Replicas = pointer.Int32Ptr(0)
+
+					_, err := updateDeployment(kubeClient, &deploy, deploy.Namespace)
+					if err != nil {
+						logger.Errorf(err, "Could not scale down deployment %s.%s", deploy.Name, deploy.Namespace)
+						errs = append(errs, err)
+					} else {
+						logger.Debugf("Deployment %s.%s scaled down", deploy.Name, deploy.Namespace)
+					}
+				}
+			}
+		}
+	}
+	return errs
+}
+
+func labelDeployment(clientSet kubernetes.Interface, name, namespace string) (*appsv1.Deployment, error) {
+	labels := map[string]string{cluster.LabelDeploymentProxless: "true"}
+
+	deploy, err := getDeployment(clientSet, name, namespace)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if deploy.Labels == nil {
+		deploy.Labels = map[string]string{}
+	}
+	deploy.Labels = utils.MergeMap(deploy.Labels, labels)
```

Comment:
comment - there's a few libs out there. if you're gonna make a utils, may look for a few full on libs that are avail. if you're a fan of linq in java/c# -> https://github.com/ahmetb/go-linq is interesting too.

Reply:
I checked online but did not find that :p
Thanks!

Reply:
One question though, how big would be the generated executable with this repo versus with a few redefined functions? Might be wrong but if I remember correctly, go import everything.